service/storage_service.cc

```diff
@@ -499,12 +499,42 @@ future<> storage_service::merge_topology_snapshot(raft_topology_snapshot snp) {
    co_await _db.local().apply(freeze(muts), db::no_timeout);
 }
 
+template<typename Builder>
```

Comment:
Given @kbr-scylla will make the query processor available soon-ish early at startup phase maybe we don't need to invest so much into the mutation builder logic  - we will be switching topology code to CQL. But the patch itself is fine of course.

Reply:
> we will be switching topology code to CQL

Huh?

Reply:
We discussed this when the centralized topology coordinator was introduced. Using the mutation builder is there only because we wanted to avoid CQL and wanted something quick.

Reply:
There was nothing preventing us from using CQL and my refactors don't affect it. Topology coordinator code sits in `storage_service` and is executed when `query_processor` is already available.

Reply:
I think I remember clearer now, the other problem was that group0 client needs a mutation.

Reply:
> I think I remember clearer now, the other problem was that group0 client needs a mutation.

This is the only reason .  We prepare raft commands here, not writing into a table.

Reply:
What we discussed is that we need to optimize loading topology to not re-read everything,  but look into a mutation in a command and figure what needs to be changed.